pkg/spire/controller.go

```diff
@@ -31,17 +32,44 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/logging"
 )
 
+func init() {
+	injection.Default.RegisterClient(withControllerClient)
+}
+
+// controllerKey is a way to associate the ControllerAPIClient from inside the context.Context
+type controllerKey struct{}
+
+// GetControllerAPIClient extracts the ControllerAPIClient from the context.
+func GetControllerAPIClient(ctx context.Context) ControllerAPIClient {
```

Comment:
Doing some testing here. Seems like I am running into the `untyped == nil`. Do I need to setup the context in some way for this to work?

Reply:
hmm interesting, yea maybe it requires setting up the context with the injection library.